src/helpers/eventCache.ts

```diff
@@ -0,0 +1,31 @@
+import { EventCache, EventDateProvider } from './EventCache';
+import { ThreadEvents } from './ThreadEvents';
+
+export class BaseEvents {
+    public threads: ThreadEvents;
```

Comment:
If we move away from replacement, we might need an LRU strategy

Reply:
Great idea. I've sketched this out here. https://www.notion.so/next/3084a3234